File: src/lto-win32.cc
#include "mold.h"
#include "lto.h"

namespace mold {

template <typename E>
ObjectFile<E> *read_lto_object(Context<E> &ctx, MappedFile *mf) {
  Fatal(ctx) << "LTO is not supported on Windows";
}

template <typename E>
std::vector<ObjectFile<E> *> run_lto_plugin(Context<E> &ctx) {
  return {};
}

template <typename E>
void lto_cleanup(Context<E> &ctx) {}

using E = MOLD_TARGET;

template ObjectFile<E> *read_lto_object(Context<E> &, MappedFile *);
template std::vector<ObjectFile<E> *> run_lto_plugin(Context<E> &);
template void lto_cleanup(Context<E> &);

} // namespace mold

File: src/lto_win32.rs
//! Native Windows LTO support.

use crate::context::Context;
use crate::fatal;
use crate::input_files::ObjectFile;
use crate::mapped_file::MappedFile;
use crate::target::Target;

pub fn read_lto_object<E: Target>(
    _ctx: &mut Context<E>,
    _mf: &'static MappedFile,
    _archive_name: &'static std::path::Path,
) -> Option<ObjectFile<E>> {
    fatal!("LTO is not supported on Windows");
}

pub fn run_plugin<E: Target>(_ctx: &mut Context<E>) {}

pub fn cleanup() {}
